src/presets/src/Aircraft/AircraftPreset.h

```diff
@@ -0,0 +1,128 @@
+// Copyright (c) 2022 FlyByWire Simulations
+// SPDX-License-Identifier: GPL-3.0
+
+#pragma once
+
+#include <MSFS\Legacy\gauges.h>
+#include <MSFS\MSFS.h>
+#include <MSFS\MSFS_Render.h>
+#include <SimConnect.h>
+
+#include <functional>
+#include <iostream>
+#include <map>
+#include <vector>
+
+#include "Units.h"
+#include "../inih/ini.h"
+#include "AircraftProcedures.h"
+
+using namespace std;
+
+/**
+ * Class for handling aircraft presets.
+ */
+class AircraftPreset {
+private:
+  Units* m_Units{};
+
+  // Sim LVAR IDs
+  ID LoadAircraftPresetRequest{};
+  ID ProgressAircraftPreset{};
+
+  // Simvar light variables
+  ENUM SimOnGround{};
+
+  bool isInitialized = false;
+
+  // Procedures
+  AircraftProcedures procedures{};
+
+  // current procedure ID
+  int64_t currentProcedureID = 0;
+  // current procedure
+  vector<ProcedureStep>* currentProcedure = nullptr;
+  // flag to signal that a loading process is ongoing
+  bool loadingIsActive = false;
+  // in ms
+  double currentLoadingTime = 0.0;
+  // time for next action in respect to currentLoadingTime
+  double currentDelay = 0;
+  // step number in the array of steps
+  uint64_t currentStep = 0;
+
+public:
+  /**
+   * Creates an instance of the LightPreset class.
+   * @param simVars pointer to the LightSimVars object for reading and writing
+   * the simulation variables.
+   */
+  AircraftPreset() {
+    m_Units = new Units();
```

Comment:
Wouldn't be better to write all definitions of the functions in the CPP-files?
A mix of definition-location is a bit confusing

Reply:
Not really - short ones like constructor/destructors, getter, setter could remain in the h file.
Although not relevant in this use case definitions in h file can be inlined by compiler.

I have moved the constructor/destructor but left the getter and setters. 
Ok?

Reply:
I am not sure how clang handles inline-functions, but MSVC requires __inline to evaluate a possible function-inlining.
After a short check in google does it look like that clang requires the 'inline', too.

Reply:
Afaik "inline" is just a hint for the compiler to consider to inline with a higher priority. All optimizing compilers will inline very short functions as adding a function call instead of inlining for a simple getter/setter would just add overhead.
https://en.cppreference.com/w/cpp/language/inline

But as I said - that is not relevant for this simple use case and performance is not a big aspect here neither. 

I simply think that definitions for simple getters/setters (and constructors/destructors) are better done in the header file to keep clutter out of the cpp file and also it is more convenient. There might of course be different conventions with in different groups of developers. Let's continue this discussion on Discord 😄

Reply:
> I am not sure how clang handles inline-functions, but MSVC requires __inline to evaluate a possible function-inlining. After a short check in google does it look like that clang requires the 'inline', too.

It really shouldn't at -O >= 1